include/gepetto/viewer/macros.h

```diff
@@ -9,7 +9,7 @@
 #ifndef SCENEVIEWER_MACROS_HH
 #define SCENEVIEWER_MACROS_HH
 
-#if __cplusplus > 199711L
+#if __cplusplus > 199711L && !defined(URDFDOM_BOOST_SHARED_PTR)
```

Comment:
I think it is not a good idea to use boost::shared_ptr or std::shared_prt depending on the version of C++.
As done in https://github.com/stack-of-tasks/pinocchio, you must only take care of this inside the URDF parser, not inside the whole program.

Reply:
> I think it is not a good idea to use boost::shared_ptr or std::shared_ptr depending on the version of C++.

I do not think it hurts (in either way). What hurts is the different behavior of the whole program depending on urdfdom version. As @jcarpent suggests, the problem related to urdfdom should be handled in the parser.

> you must only take care of this inside the URDF parser

i.e. in `src/urdf-parser.cpp`

Reply:
I agree, this should only affect the urdfparser :)